tests/test_name_suggestion_index.py

```diff
@@ -27,3 +27,26 @@ def test_iter_nsi():
     i = matches[0]
     assert i["displayName"] == "Greggs"
     assert i["tags"]["amenity"] == "fast_food"
+
+
+def test_generate_keys_from_nsi_attributes():
+    nsi = NSI()
+    matches = list(nsi.iter_nsi("Q38076"))
+    keys_to_find = [
+        ("mcdonalds", "McDonaldsSpider"),
+        ("mcdonalds_fr", "McDonaldsFRSpider"),
+        ("mai_dang_lao_mcdonalds_hk_mo", "MaiDangLaoMcDonaldsHKMOSpider"),
```

Comment:
`MaiDangLaoMcDonaldsHKMOSpider` should now be `MaiDangLaoMcdonaldsHKMOSpider`... I think... after the recent changes of #8829 to automatically fix/set spider class names based on the spider `name` field.